backend/src/test/java/com/funeat/review/persistence/ReviewRepositoryTest.java

```diff
@@ -1,124 +1,156 @@
-
-    @Autowired
-    private CategoryRepository categoryRepository;
-
-    @Autowired
-    private ProductRepository productRepository;
-
-    @Autowired
-    private ReviewRepository reviewRepository;
-
-    @Test
-    void 상품에_달린_리뷰의_숫자를_반환한다() {
-        // given
-        Member member = memberRepository.save(new Member("test", "image.png", "1"));
-        Category category = categoryRepository.save(new Category("간편식사", CategoryType.FOOD));
-        Product product1 = productRepository.save(new Product("삼각김밥", 1000L, "image.png", "맛있는 삼각김밥", category));
-        Product product2 = productRepository.save(new Product("라면", 2000L, "image.png", "맛있는 라면", category));
 
-        reviewRepository.save(new Review(member, product1, "review.png", 4L, "이 삼각김밥은 최고!!", true));
-        reviewRepository.save(new Review(member, product1, "review.png", 3L, "이 삼각김밥은 별로", false));
-        reviewRepository.save(new Review(member, product1, "review.png", 4L, "이 삼각김밥은 쏘쏘", true));
-        reviewRepository.save(new Review(member, product2, "review.png", 3L, "이 라면은 맛있다", true));
-
-        // when
-        assertThat(reviewRepository.countByProduct(product1)).isEqualTo(3);
-        assertThat(reviewRepository.countByProduct(product2)).isEqualTo(1);
+@SuppressWarnings("NonAsciiCharacters")
+class ReviewRepositoryTest extends RepositoryTest {
+
+    @Nested
+    class countByProduct_테스트 {
+
+        @Nested
+        class 성공_테스트 {
+
+            @Test
+            void 상품의_리뷰_수를_반환한다() {
+                // given
+                final var member1 = 멤버_멤버1_생성();
+                final var member2 = 멤버_멤버2_생성();
+                final var member3 = 멤버_멤버3_생성();
+                복수_멤버_저장(member1, member2, member3);
+
+                final var category = 카테고리_간편식사_생성();
+                단일_카테고리_저장(category);
+
+                final var product1 = 상품_삼각김밥_가격1000원_평점_2점_생성(category);
+                final var product2 = 상품_삼각김밥_가격2000원_평점_3점_생성(category);
+                복수_상품_저장(product1, product2);
+
+                final var review1_1 = 리뷰_이미지test4_평점4점_재구매O_생성(member1, product1, 0L);
+                final var review1_2 = 리뷰_이미지test3_평점3점_재구매X_생성(member2, product1, 0L);
+                final var review1_3 = 리뷰_이미지test4_평점4점_재구매O_생성(member3, product1, 0L);
+                final var review2_1 = 리뷰_이미지test3_평점3점_재구매O_생성(member1, product2, 0L);
+                복수_리뷰_저장(review1_1, review1_2, review1_3, review2_1);
+
+                final var expected1 = 3;
+                final var expected2 = 1;
+
+                // when
+                final var actual1 = reviewRepository.countByProduct(product1);
+                final var actual2 = reviewRepository.countByProduct(product2);
+
+                // then
+                assertSoftly(softAssertions -> {
+                    softAssertions.assertThat(actual1).isEqualTo(expected1);
+                    softAssertions.assertThat(actual2).isEqualTo(expected2);
+                });
```

Comment:
```suggestion
                assertSoftly(softAssertions -> {
                    softAssertions.assertThat(actual1).isEqualTo(3);
                    softAssertions.assertThat(actual2).isEqualTo(1);
                });
```
이런식으로 테스트에 쓰이는 숫자는 그냥 그대로 사용하는건 어떨까요?!
변수로 빼면 해당 변수를 찾으러 코드를 거슬러 올라가야해서 then절의 가독성이 조금 떨어지는 것 같아서요..!

Reply:
전부 수정 완료